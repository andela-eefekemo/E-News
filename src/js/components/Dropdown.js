import React from 'react';
import PropTypes from 'prop-types';

const Dropdown = (props) => {
  const { value, text } = props;
  return (
    <option value={value}>{text}</option>
  );
};

Dropdown.defaultProps = {
  value: '',
  text: '',
};

Dropdown.propTypes = {
  value: PropTypes.String,
  text: PropTypes.String,
};

export default Dropdown;
