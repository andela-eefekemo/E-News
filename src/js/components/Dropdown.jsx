import React from 'react';
import PropTypes from 'prop-types';

/**
 * @param {any} props
 * @returns
 */
const Dropdown = (props) => {
  const { value, text } = props;
  return <option value={value}>{text}</option>;
};

Dropdown.defaultProps = {
  value: '',
  text: '',
};

Dropdown.propTypes = {
  value: PropTypes.string,
  text: PropTypes.string,
};

export default Dropdown;
