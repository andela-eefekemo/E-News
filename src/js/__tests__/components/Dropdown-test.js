import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Dropdown from '../../components/Dropdown';

test('Dropdown component', () => {
  const component = shallow(<Dropdown />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
