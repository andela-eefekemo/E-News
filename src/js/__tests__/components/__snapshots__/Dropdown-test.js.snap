// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown component 1`] = `
<option
  value=""
/>
`;
